pkg/api/controller.go

```diff
@@ -2134,21 +2135,50 @@ func (c *Controller) UploadObject(w http.ResponseWriter, r *http.Request, reposi
 		allowOverwrite = false
 	}
 
-	// write the content
-	file, handler, err := r.FormFile("content")
-	if errors.Is(err, http.ErrMissingFile) {
-		writeError(w, r, http.StatusInternalServerError, fmt.Errorf("multipart uploads missing key 'content': %w", err))
-		return
-	}
+	// read request body parse multipart for "content" and upload the data
+	mt, p, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
 	if err != nil {
 		writeError(w, r, http.StatusInternalServerError, err)
 		return
 	}
-	defer func() { _ = file.Close() }()
-	contentType := handler.Header.Get("Content-Type")
-	address := c.PathProvider.NewPath()
-	blob, err := upload.WriteBlob(ctx, c.BlockAdapter, repo.StorageNamespace, address, file, handler.Size, block.PutOpts{StorageClass: params.StorageClass})
-	if err != nil {
+	if !strings.HasPrefix(mt, "multipart/") {
+		writeError(w, r, http.StatusInternalServerError, http.ErrNotMultipart)
+		return
+	}
```

Comment:
Not sure this works for _any_ multipart Content-Type, perhaps we should limit ourselves to a list of known types?  (Package http probably limits itself in some way, so if we do the same would we be safe?)

Reply:
Good point - I'll limit it to "multipart/form-data". The Go's request implementation support `form-data` and optional `mixed`.

Reply:
> Also, the current implementation will be dangerous if we ever add another part to UploadObject in swagger.yml. Let's add a comment on its definition _not_ to do that!

Currently we added the information in the swagger description: `Only a single file per upload which must be named "content".`